hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsBlock.java

```diff
@@ -55,7 +57,9 @@ public class AbfsBlock {
      * @return String representing the block ID generated.
      */
     private String generateBlockId(long position) {
-        String blockId = String.format("%d", position);
+        String streamId = this.outputStream.getStreamID();
+        String streamIdHash = Integer.toString(streamId.hashCode());
+        String blockId = String.format("%d_%s", position, streamIdHash);
```

Comment:
are we confident that it will be under `BLOCK_ID_LENGTH` (60)?

Reply:
Yes should work as max size of position can be 8 bytes

Reply:
Adding to it:
The hashcode of a string is an integer value that represents the content of the string in a compact way. The hashcode is not related to the size of the string, but to the characters in it. [The hashcode can have any value between Integer.MIN_VALUE and Integer.MAX_VALUE, which are -2147483648 and 2147483647 respectively](https://stackoverflow.com/questions/31027909/string-comparison-using-hashcode-and-length)


So, in case our function becomes like:
```
private String generateBlockId(long position) {
        String streamId = this.outputStream.getStreamID();
        String streamIdHash = Integer.toString(streamId.hashCode());
        String blockId = String.format("%d_%s", Long.MAX_VALUE, Integer.MAX_VALUE);
        byte[] blockIdByteArray = new byte[BLOCK_ID_LENGTH];
        System.arraycopy(blockId.getBytes(), 0, blockIdByteArray, 0, Math.min(BLOCK_ID_LENGTH, blockId.length()));
        return new String(Base64.encodeBase64(blockIdByteArray), StandardCharsets.UTF_8);
    }
```

`blockId.length()` would be 19 + 1 + 10 = 30.